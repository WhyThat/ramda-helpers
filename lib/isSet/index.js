import { complement } from 'ramda';
import isNilOrEmpty from '../isNilOrEmpty';

/**
 * Returns `true` for value that are defined and not empty, `false` otherwise
 * @name isSet
 * @memberof RamdaHelper
 * @see isNilOrEmpty
 * @param {*} value
 * @returns {Boolean}
 * @example
 *
 * isSet([]) //=> false
 * isSet({}) //=> false
 * isSet('') //=> false
 * isSet(null) //=> false
 * isSet(false) //=> true
 * isSet('test') //=> true
 * isSet(['test']) //=> true
 * isSet({ a: 1 }) //=> true
 */
export default complement(isNilOrEmpty);
