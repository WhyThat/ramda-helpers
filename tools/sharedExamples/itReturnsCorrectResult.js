export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
