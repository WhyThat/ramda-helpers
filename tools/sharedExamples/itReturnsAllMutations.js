import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
