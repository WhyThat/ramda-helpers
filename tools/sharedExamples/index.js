
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
