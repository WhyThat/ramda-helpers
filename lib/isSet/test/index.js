import { itReturnsAllMutations } from '../../../tools/sharedExamples';
import isSet from '../';

describe('isSet', () => {
  const mutations = [{
    name: 'With null',
    props: null,
  }, {
    name: 'With undefined',
    props: undefined,
  }, {
    name: 'With empty string',
    props: '',
  }, {
    name: 'With empty array',
    props: [],
  }, {
    name: 'With empty object',
    props: {},
  }, {
    name: 'With an non-empty array',
    props: ['non-Empty'],
  }, {
    name: 'With an non-empty object',
    props: { a: '' },
  }];

  itReturnsAllMutations(isSet, mutations);
});
