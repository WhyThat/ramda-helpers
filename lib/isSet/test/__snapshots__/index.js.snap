// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isSet With an non-empty array returns correct result 1`] = `true`;

exports[`isSet With an non-empty object returns correct result 1`] = `true`;

exports[`isSet With empty array returns correct result 1`] = `false`;

exports[`isSet With empty object returns correct result 1`] = `false`;

exports[`isSet With empty string returns correct result 1`] = `false`;

exports[`isSet With null returns correct result 1`] = `false`;

exports[`isSet With undefined returns correct result 1`] = `false`;
